===================================================================
RCS file: /cvsroot/abook/abook/filter.c,v
retrieving revision 1.29
retrieving revision 1.30
diff -u -r1.29 -r1.30
--- abook/abook/filter.c	2004/01/23 07:14:32	1.29
+++ abook/abook/filter.c	2004/03/25 18:19:38	1.30
@@ -1,6 +1,6 @@
 
 /*
- * $Id: abook-0.5.2-filter.diff,v 1.1 2004/03/25 21:49:43 rizzo Exp $
+ * $Id: abook-0.5.2-filter.diff,v 1.1 2004/03/25 21:49:43 rizzo Exp $
  *
  * by JH <jheinonen@users.sourceforge.net>
  *
@@ -618,6 +618,7 @@
 mutt_read_line(FILE *in, char **alias, char **rest)
 {
 	char *line, *ptr, *tmp;
+	size_t alias_len;
 
 	if( !(line = ptr = getaline(in)) )
 		return 1; /* error / EOF */
@@ -640,13 +641,16 @@
 	while( ! ISSPACE(*ptr) )
 		ptr++;
 
-	if( (*alias = (char *)malloc(ptr - tmp)) == NULL) {
+	/* includes also the trailing zero */
+	alias_len = (size_t)(ptr - tmp + 1);
+
+	if( (*alias = (char *)malloc(alias_len)) == NULL) {
 		free(line);
 		return 1;
 	}
 
-	strncpy(*alias, tmp, ptr - tmp - 1);
-	*(*alias + (ptr - tmp - 1)) = 0;
+	strncpy(*alias, tmp, alias_len - 1);
+	*(*alias + alias_len - 1) = 0;
 
 	while(ISSPACE(*ptr))
 		ptr++;
